samples/excel/30-events/events-comment-event-handler.yaml

```diff
@@ -0,0 +1,195 @@
+                // Retrieve the added comment.
+                let addedComment = context.workbook.comments.getItem(event.commentDetails[0].commentId);
+
+                // Load the added comment's data.
+                addedComment.load(["content", "authorName", "creationDate"]);
+
+                await context.sync();
+
+                // Print out the added comment's data.
+                console.log(`A comment was added:`);
+                console.log(`    ID: ${event.commentDetails[0].commentId}`);
+                console.log(`    Comment content:${addedComment.content}`);
+                console.log(`    Comment author:${addedComment.authorName}`);
+                console.log(`    Creation date:${addedComment.creationDate}`);
+            });
+        }
+
+        async function commentChanged(event: Excel.CommentChangedEventArgs) {
+            await Excel.run(async (context) => {
+                // Retrieve the changed comment.
+                let changedComment = context.workbook.comments.getItem(event.commentDetails[0].commentId);
+
+                // Load the changed comment's data.
+                changedComment.load(["content", "authorName", "creationDate"]);
+
+                await context.sync();
+
+                // Print out the changed comment's data.
+                console.log(`A comment was changed:`);
+                console.log(`    ID: ${event.commentDetails[0].commentId}`);
+                console.log(`    Updated comment content: ${changedComment.content}`);
+                console.log(`    Comment author: ${changedComment.authorName}`);
+                console.log(`    Creation date: ${changedComment.creationDate}`);
+            });
+        }
+
+        async function commentDeleted(event: Excel.CommentDeletedEventArgs) {
+            await Excel.run(async (context) => {
+                // Print out the deleted comment's ID.
+                console.log(`A comment was deleted:`);
+                console.log(`    ID: ${event.commentDetails[0].commentId}`);
+            });
+        }
+
+        async function addCommentToSelectedCell() {
+            await Excel.run(async (context) => {
+                const selectedRange = context.workbook.getSelectedRange();
+
+                // Note that an InvalidArgument error will be thrown if multiple cells are selected.
+                context.workbook.comments.add(selectedRange, "TODO: add headers here.");
+
+                await context.sync();
+            });
+        }
+
+        async function addCommentToCell() {
+            await Excel.run(async (context) => {
+                const sheet = context.workbook.worksheets.getItem("Comments");
+
+                // Note that an InvalidArgument error will be thrown if multiple cells are passed to `comment.add`.
+                sheet.comments.add("A2", "TODO: add data.");
+                await context.sync();
+            });
+        }
+
+        async function editComment() {
```

Comment:
Consider as a future work item adding comment replies to this sample. I think it would be interesting to show how adding, deleting, and editing replies triggers these events.